Fortunate archeology: componentize Report Abuse popover

modernserf opened the pull request:
## Links
* Remix link: https://fortunate-archeology.glitch.me/

## GIF/Screenshots:
![Screen Shot 2019-05-21 at 4 48 55 PM](https://user-images.githubusercontent.com/938722/58129852-b755e680-7be8-11e9-9f6c-4d6ca3c6ab53.png)
![Screen Shot 2019-05-21 at 4 27 19 PM](https://user-images.githubusercontent.com/938722/58129864-ba50d700-7be8-11e9-8957-9037419aa5a9.png)
![Screen Shot 2019-05-21 at 4 24 34 PM](https://user-images.githubusercontent.com/938722/58129870-bd4bc780-7be8-11e9-9e76-694536c0765f.png)

## Changes:
* add topLeft & topRight to popover alignments
* style refinements on PopoverInfo, PopoverTitle, InfoDescription
* componentize ReportAbusePop

## How To Test:
- visit https://fortunate-archeology.glitch.me/
  * clicking Report Abuse should open the Report Abuse window
  * your email address should be validated as you type
  * the report should be validated as you type
  * submitting a report should create a ticket in `#abuse`
 - log in with your persistent user token
  * the Report Abuse flow should be the same, except it shows your username instead of a field for an email address

clottman commented:
Old: 
![image](https://user-images.githubusercontent.com/4480480/58192463-6dbbd900-7c86-11e9-8e83-2fb20f3b6e74.png)

New:
![image](https://user-images.githubusercontent.com/4480480/58192503-7e6c4f00-7c86-11e9-8862-b11eea906b0c.png)

In addition to visual changes, old validates that reason is different from the default reason: 
![image](https://user-images.githubusercontent.com/4480480/58192536-8d530180-7c86-11e9-8631-64395b392660.png)

The new one does not.  
![image](https://user-images.githubusercontent.com/4480480/58192544-93e17900-7c86-11e9-8ada-0c357915a51a.png)

modernserf commented:
![Screen Shot 2019-05-23 at 10 27 17 AM](https://user-images.githubusercontent.com/938722/58261423-4a586300-7d46-11e9-81a9-7c6611db39b3.png)
- added validation on blur
- updated textarea style
